src/sql/src/plan/statement/ddl.rs

```diff
@@ -2079,9 +2084,125 @@ pub fn plan_create_materialized_view(
+                        // this is not true anymore, because months have variable lengths.
+                        // See `Timestamp::round_up`.
+                        sql_bail!("REFRESH interval must not involve units larger than days");
+                    }
+
+                    let mut aligned_to = match aligned_to {
+                        Some(aligned_to) => aligned_to,
+                        None => {
+                            soft_panic_or_log!(
+                                "ALIGNED TO should have been filled in by purification"
+                            );
+                            sql_bail!(
+                                "INTERNAL ERROR: ALIGNED TO should have been filled in by purification"
+                            )
+                        }
+                    };
+
+                    // Desugar the `aligned_to` expression
+                    transform_ast::transform(scx, &mut aligned_to)?;
+
+                    let ecx = &ExprContext {
+                        qcx: &QueryContext::root(scx, QueryLifetime::OneShot),
+                        name: "REFRESH EVERY ... ALIGNED TO",
+                        scope: &Scope::empty(),
+                        relation_type: &RelationType::empty(),
+                        allow_aggregates: false,
+                        allow_subqueries: false,
+                        allow_parameters: false,
+                        allow_windows: false,
+                    };
+                    let aligned_to_hir = plan_expr(ecx, &aligned_to)?.cast_to(
+                        ecx,
+                        CastContext::Assignment,
+                        &ScalarType::MzTimestamp,
+                    )?;
+                    // (mz_now was purified away to a literal earlier)
+                    let aligned_to_const = aligned_to_hir
+                        .into_literal_mz_timestamp()
+                        .ok_or_else(|| PlanError::InvalidRefreshEveryAlignedTo)?;
+
+                    refresh_schedule.everies.push(RefreshEvery {
+                        interval: interval.duration()?,
+                        aligned_to: aligned_to_const,
+                    });
+                }
+            }
+        }
+
+        if on_commits_seen > 1 {
+            sql_bail!("REFRESH ON COMMIT cannot be specified multiple times");
+        }
+        if on_commits_seen > 0 && refresh_schedule != RefreshSchedule::empty() {
+            sql_bail!("REFRESH ON COMMIT is not compatible with any of the other REFRESH options");
+        }
+        // Note: Seeing no REFRESH options at all (not even REFRESH ON COMMIT) should be acceptable:
+        // even though purification inserts REFRESH ON COMMIT if no other REFRESH option was given,
+        // we can't rely on this behavior in planning, because this won't happen for old
+        // materialized views that were created before this feature was introduced.
```

Comment:
Don't block on this, but .. won't the migration ensure that even old matviews get `REFRESH ON COMMIT` before they ever hit planning?

Reply:
Ah yes, thx, I forgot to delete this comment when I added the migration. I'll delete it before merging the PR.